### ISSUE TITLE
Producing calibrated synthetic data for testing use (instead of training) - i.e. moving 1 degree of rotation per image

### ISSUE THREAD

**Issue by lessw2020** [Unity-Technologies/com.unity.perception#198]
Is there a way to fix the placement of an item on each image such that I could produce a calibrated test set (rather than a randomized set)?
Example - we want to be able to take a given prefab, place it in the center every time, and then rotate it 1 degree from 0 - 90 on an axis, and thus produce a calibrated set of 0 - 90 degree to test where our current models break...and repeat that for each axis to simulate various perspective skew and test where our models fall off in detection accuracy.
Similarly, place the object in a grid on the image of upper left, middle upper, upper right, etc. and create a calibrated test set of the item being placed all throughout a given image.  testing would use that to verify the range where in the image the items must be placed before accuracy falls off.
Again the idea is to produce calibration sets for testing to use rather than making these manually..so kind of the inverse of randomizing for training. :) 
I have some idea how to code this but any details would be appreciated!

**Comment by sleal-unity** [comment]
Hi Less,

The general way to do this is to create a randomizer that utilizes the current iteration index to rotate or place objects in a deterministic way. For example, if you wanted to generate 90 frames, where a central GameObject is rotated 90 degrees every frame, you could do something like this:

```csharp
using System;
using UnityEngine;
using UnityEngine.Perception.Randomization.Randomizers;
using UnityEngine.Perception.Randomization.Scenarios;

[Serializable]
[AddRandomizerMenu("Schedulers/Rotation Scheduler")]
public class RotationScheduler : Randomizer
{
    /// <summary>
    /// The GameObject already present in the scene (not a prefab) that you want to rotate in front of the camera
    /// </summary>
    public GameObject gameObjectToRotate;
    
    /// <summary>
    /// The total rotation displacement desired by the end of the scenario
    /// </summary>
    public Vector3 totalDegreesToRotate;

    Vector3 m_DegreesToRotatePerFrame;
    Quaternion m_InitialRotation;
    
    protected override void OnCreate()
    {
        // Lookup the total number of configured scenario iterations 
        var scenarioConstants = (FixedLengthScenario.Constants)scenario.genericConstants;
        var totalNumberOfIterations = scenarioConstants.totalIterations;
        
        // Calculate how far you have to rotate per frame
        m_DegreesToRotatePerFrame = totalDegreesToRotate / totalNumberOfIterations;
        m_InitialRotation = gameObjectToRotate.transform.rotation;
    }

    protected override void OnIterationStart()
    {
        // Apply the desired rotation based on the current scenario iteration.
        // Note: This strategy is compatible with distributed execution
        var rotationOffset = Quaternion.Euler(m_DegreesToRotatePerFrame * scenario.currentIteration);
        gameObjectToRotate.transform.rotation = rotationOffset * m_InitialRotation;
    }
}
```

**Comment by lessw2020** [comment]
Thanks a bunch @sleal-unity  - exactly what I need to get going on this! 
I also think this is an interesting counterpart to using synthetic for training, but also to make validation sets to give to testing.
Much appreciated!
